pkg/trie/triedb/node.go

```diff
@@ -143,32 +150,38 @@ func inMemoryFetchedValue(value nodeValue, prefix []byte, db db.DBGetter) ([]byt
 	}
 }
 
+type NodeTypes[H hash.Hash] interface {
```

Comment:
Are we using this?

Reply:
No not yet.  I think it's good to semantically describe what the `Node` types are.